### PR TITLE
adding cloud providers choice

### PR DESCRIPTION
**Pull request by choldgraf**
this adds a short section on choosing cloud providers and navigating the different services that they often offer.

### CODE REVIEW

File: content/deploy/choose_cloud.md
Comment (yuvipanda) [inline content/deploy/choose_cloud.md]: IMO, we shouldn't recommend this as a best choice. Running on your own hardware is going to be always problematic when stuff breaks. Here's a list of things that might happen to your hardware that are hard to fix without more engineering, while very easy to on the cloud:

1. You need more RAM / CPU
2. External public IPs (this is surprisingly hard with your own hardware)
3. Backups (snapshots are easy in all providers)
4. You screw up your box, and need to start over.

All these will probably happen at some point, and when most people think they have hardware support they're always going to underestimate these costs.

We should recommend they run on a cloud provider unless they can't, in which case they should run on their own hardware.

File: content/deploy/choose_cloud.md
Comment (yuvipanda) [inline content/deploy/choose_cloud.md]:
```suggestion
  getting your students used to the vendor-specific interface rather than an open-source version.
```

File: content/deploy/choose_cloud.md
Comment (yuvipanda) [inline content/deploy/choose_cloud.md:76]: A big con is also that you don't have control over a lot of things there. Here's a small list:

1. Environment might not have the versions of packages you want installed (this is more a problem in Colab maybe? Unclear)
2. You don't have access to extensions or customizations as you want. No nbgitpuller, no canvas / LMS integration, no customizable resources, etc
3. ...

While the core reasons are one of control, we should definitely list some strong technical reasons too, despite the possibility they are gonna get out of date quickly. I also don't know how many of these are true right now.

**Review by yuvipanda — CHANGES_REQUESTED** [review]
This is a good start! How should we approach recommending particular providers? I personally think you should just use Digital Ocean unless you have access to cloud credits, but I don't think that can be a 'project' opinion...

Do you think we should mention we follow this practice at Berkeley? Or should that be someplace else?

Also, thank you very much for working on this

File: content/deploy/choose_cloud.md
Comment (lheagy) [inline content/deploy/choose_cloud.md]: AWS too?

File: content/deploy/choose_cloud.md
Comment (lheagy) [inline content/deploy/choose_cloud.md]: I would suggest framing this similar to the others with "pros" and "cons". For under-resourced instructors, using a managed "free" service may be the most feasible approach for them in the first deployment of a course. We don't need to spend much time discussing how to do this, and focus in on DIY approaches, but I think there are quite a number of instructors for whom launching their own hub is a complete non-starter. If they can get started on a service, and then transition, perhaps in the next delivery of the course to using their own deployment, then that is still a net-win.

File: content/deploy/choose_cloud.md
Comment (lheagy) [inline content/deploy/choose_cloud.md]: Does this exist anywhere yet? I haven't seen a "push button" approach by any of the cloud providers yet. Are there examples we could point to? Otherwise, this discussion is quite abstract and not necessarily helpful for an instructor looking to get started.

File: content/deploy/choose_cloud.md
Comment (lheagy) [inline content/deploy/choose_cloud.md]: The language here is potentially quite off-putting, "This should be your ..." is essentially imposing your value-system on instructors, and potentially comes across as judgemental if their priority system doesn't align with yours. I would recommend that we re-work the language to be a bit more matter-of-fact, and let the pros / cons do the work of providing a structure for forming an opinion.

**Comment by choldgraf** [comment]
ok, I removed some of the more "opinionated" statements from the "choosing cloud resources" page and update the list there a bit, also added a section elsewhere on berkeley's approach to open source. Would love to hear what @lheagy and @yuvipanda think about it!

File: content/tech/considerations.md
Comment (lheagy) [inline content/tech/considerations.md]:
```suggestion
### The Scipy stack - an open source ecosystem
```
?

**Review by lheagy — APPROVED** [review]
This looks really solid @choldgraf! The separation of the `Open source strategy` from the steps and decision points of an individual instructor relieves the concerns I had about the previous version. Thanks for pushing on this - it will be a valuable resource for the community!

**Comment by lheagy** [comment]
One other quick question: do we have a pointer somewhere to guide people when to use TLJH vs. Z2JH? giving a rough "if your class is < 100?" (cc @yuvipanda) - then deploying TLJH is recommended (v. easy on digital ocean, for example), and if larger, then Z2JH. This might already be somewhere else, in the docs?

**Comment by choldgraf** [comment]
Good question - we do have that covered in the next section here: http://data8.org/zero-to-data-8/deploy/setup_jupyterhub.html#deploying-jupyterhub-in-the-cloud

I just pushed a commit that tries to clean up the section header and make it a bit more clear what the section is about.

@yuvipanda are you happy with the language right now? (you've got a "change requested" flag on this PR still)

**Comment by choldgraf** [comment]
@yuvipanda I believe that I addressed your issue before, so I'm gonna merge this one and feel free to let me know if you' like me to change something moving forward!

**Pull request merged**
